### PR TITLE
editline: fix cross-build to iOS + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by theirix — APPROVED** [review]

File: recipes/editline/all/conanfile.py [hunk 1]
```diff
@@ -47,13 +39,23 @@ def configure(self):
...
+            self.requires("termcap/1.3.1")
+        elif self.options.terminal_db == "ncurses":
+            self.requires("ncurses/6.2")
+
+    def validate(self):
         if self.settings.os == "Windows":
             raise ConanInvalidConfiguration("Windows is not supported by libedit (missing termios.h)")
+        if self.options.terminal_db == "tinfo":
+            # TODO - Add tinfo when available
+            raise ConanInvalidConfiguration("tinfo is not (yet) available on CCI")
```
Comment (madebr) [inline recipes/editline/all/conanfile.py:54]: `tinfo` is part of ncurses.
ncurses can be built with tinfo built inside `libncurses.so`, or separately as `libtinfo.so` and `libncruses.so`.
https://github.com/conan-io/conan-center-index/blob/6704c4d14b626b69e74ff5f9c79ed235316fb8b7/recipes/ncurses/all/conanfile.py#L27

Comment (SpaceIm) [reply recipes/editline/all/conanfile.py:54]: I see. I didn't change this part, juste moved the verification in validate(). It can be fixed in another PR.

Comment (SpaceIm) [reply recipes/editline/all/conanfile.py:54]: What do you sugget? Remove tinfo in possibles values of `terminal_db` option?

Comment (madebr) [reply recipes/editline/all/conanfile.py:54]: I would remove (or deprecate) the option.
And pass `tinfo` if `ncurses:with_tinfo` is True, else `ncurses`.

Comment (SpaceIm) [reply recipes/editline/all/conanfile.py:54]: editline doesn't have such option. It tests existence of ncurses, then curses, then termcap, then tinfo, that's all.

```autoconf
AC_CHECK_LIB(ncurses, tgetent,,
  [AC_CHECK_LIB(curses, tgetent,,
    [AC_CHECK_LIB(termcap, tgetent,,
      [AC_CHECK_LIB(tinfo, tgetent,,
        [AC_MSG_ERROR([libncurses, libcurses, libtermcap or libtinfo is required!])]
      )]
    )]
  )]
)
```

Comment (madebr) [reply recipes/editline/all/conanfile.py:54]: I guess that will work as the `tgetent` symbol will not be found in `ncurses`.
So only removing/deprecating the option value is enough.

Comment (SpaceIm) [reply recipes/editline/all/conanfile.py:54]: we can remove it without deprecation, it was an invalid configuration anyway.

Comment (madebr) [reply recipes/editline/all/conanfile.py:54]: good call :)

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**